pyracf/common/utilities/logger.py

```diff
@@ -149,30 +149,97 @@ def __redact_request_dictionary(
+        with ( and ), but can conceivably contain any characters, but a negative lookbehind
+        is used to look for any escape \ character that would indicate the matching of the
+        ( and ) is otherwise a coincidence.
+
+        In all replacement expressions, the variable amounts of whitespace are captured so that
+        they can be preserved by this redaction operations.
+
         """
         asterisks = "********"
+        is_bytes = False
         if isinstance(input_string, bytes):
-            asterisks = asterisks.encode("cp1047")
-        pre_keyword = input_string[:start_ind]
-        post_keyword = end_pattern.join(input_string[start_ind:].split(end_pattern)[1:])
-        return pre_keyword + asterisks + end_pattern + post_keyword
+            input_string = input_string.decode("cp1047")
+            is_bytes = True
+        quoted = re.search(
+            rf"{trait_key.upper()}( +){{0,}}\(\'.*?(?<!\')\'\)", input_string
+        )
+        nested = re.search(
+            rf"{trait_key.upper()}( +){{0,}}\([^)]*\(.*\)( +){{0,}}\)", input_string
+        )
+        if quoted is not None:
+            input_string = re.sub(
+                rf"{trait_key.upper()}( +){{0,}}\(\'.*?(?<!\')\'\)",
+                rf"{trait_key.upper()}\1('{asterisks}')",
+                input_string,
+            )
+        else:
+            if nested is not None:
+                input_string = re.sub(
+                    rf"{trait_key.upper()}( +){{0,}}\([^)]*\(.*\)( +){{0,}}\)",
+                    rf"{trait_key.upper()}\1({asterisks})",
+                    input_string,
+                )
+            else:
+                input_string = re.sub(
+                    rf"{trait_key.upper()}( +){{0,}}\(.*?(?<!\\)\)",
+                    rf"{trait_key.upper()}\1({asterisks})",
+                    input_string,
+                )
+        if is_bytes:
+            return input_string.encode("cp1047")
+        return input_string
 
     def redact_request_xml(
         self,
         xml_string: Union[str, bytes],
         secret_traits: dict,
     ) -> Union[str, bytes]:
-        """
+        r"""
         Redact a list of specific secret traits in a request xml string or bytes object.
         Based the following xml pattern:
             '<xmltag attribute="any">xml value</xmltag>'
         This function also accounts for any number of arbitrary xml attributes.
+
+        This function employs the following regular expression:
+        {xml_key}(.*)>.*<\/{xml_key} - Designed to match the above pattern by starting and ending
+        with the xmltag string as shown, but the starting tage allows for any characters between
+        "xmltag" and the > character to allow for the attribute specification shown above. This
+        results in the starting of the xml as {xml_key}(.*)> and the ending as <\/{xml_key}.
+        The miscellaneous characters are "captured" as a variable and preserved by the
```

Comment:
Change to: `The characters between the xml tags are "captured"`.

Reply:
Change made.